--- a/swf/SWF_ScriptObject.cpp
+++ b/swf/SWF_ScriptObject.cpp
@@ -525,7 +525,7 @@ idSWFScriptObject::swfNamedVar_t* idSWFScriptObject::GetVariable( const char* na
 	int hash = idStr::Hash( name ) & ( VARIABLE_HASH_BUCKETS - 1 );
 	for( int i = variablesHash[hash]; i >= 0; i = variables[i].hashNext )
 	{
-		if( variables[i].name == name )
+		if((i < variables.Num()) && (variables[i].name == name))
 		{
 			return &variables[i];
 		}
